Wikipedia/Code/ArticleViewController+FindInPage.swift

```diff
@@ -75,6 +75,7 @@ extension ArticleViewController {
         self.findInPage.view?.removeFromSuperview()
         self.view.layoutSubviews()
         self.findInPage.view = nil
+        self.view.becomeFirstResponder()
```

Comment:
What a weird bug!

I still see an issue when I do this:

1. Tap featured article
2. Tap find, type "the". If in simulator, ensure simulator keyboard appears.
3. Clear search field, close find.
4. Keyboard dismisses. Interestingly tapping a link that was under the keyboard doesn't work, but tapping a link near the top does work.

It does look a bit fishy that you are changing responders here, then above on line 71 you are also messing with responders in the completion block. This line may get run before line 71. It might work better if all responder changes happen within that completion block, but I'm not having any luck so far in my tinkering.

Reply:
Update: This is working a little better for me, but you should double-check it:

```
    func hideFindInPage(_ completion: (() -> Void)? = nil) {
        resetFindInPage {
            self.findInPage.view?.hide()
            self.findInPage.view?.removeFromSuperview()
            self.findInPage.view = nil
            self.resignFirstResponder()
            completion?()
        }
    }
```

I noticed `showFindInPage()` above has a `becomeFirstResponder()` call, so I'm balancing it out here with a `self.resignFirstResponder()`. I'm not sure if that's helping or not, or if I'm just randomly hitting better scenarios.

Reply:
This solution is working better for item #4 you described here. My previous attempt was working for the scenarios I tested, so it was a little tricky to debug. The responder being altered inside and outside that closure might be the cause of this behaviour here. I'll maintain this version and have QA test for other possibilities. Thanks for the help!